### PR TITLE
Removing deprecated code for _model_to_fit_params

### LINKED ISSUE #15455: Removal of deprecated code before 6.0 feature freeze
### What is the problem this feature will solve?

[Feature freeze for `astropy` 6.0 is scheduled to be in the end of this month,](https://github.com/astropy/astropy/wiki/Release-Calendar) but there's still quite a lot of deprecated functionality in the codebase. Note that [according to APE 2 deprecation warnings should be visible to users for at least a year.](https://github.com/astropy/astropy-APEs/blob/main/APE2.rst#releases-and-backwards-compatibility) From the release calendar it can be seen that [`astropy` 5.1 was released on 2022-05-25 and 5.2 on 2022-12-12](https://github.com/astropy/astropy/wiki/Release-Calendar), so in 6.0 it is only permitted to remove code that was deprecated in `astropy` 5.1 or earlier.

### Describe the desired outcome

The following list contains items that were deprecated in `astropy` 5.1 or earlier as of 1ae27848e23910d6f1383ee47e02a53d8d73b3b3 (i.e. `main` at the time of opening this issue). They are ordered by the `astropy` version they were deprecated in.

- [x] https://github.com/astropy/astropy/pull/15462 https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/timeseries/sampled.py#L156-L157
- [x] https://github.com/astropy/astropy/pull/15463 https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/coordinates/builtin_frames/galactocentric.py#L269-L271
- [x] https://github.com/astropy/astropy/pull/15464 https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/wcs/wcs.py#L2109-L2110
- [x] https://github.com/astropy/astropy/pull/15466 https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/config/configuration.py#L50-L53
- [x] https://github.com/astropy/astropy/pull/15466 https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/config/configuration.py#L777-L782
- [x] https://github.com/astropy/astropy/pull/15467 https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/cosmology/realizations.py#L112-L114
- [x] https://github.com/astropy/astropy/pull/15468 https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/utils/introspection.py#L105-L106
- [x] https://github.com/astropy/astropy/pull/15470 https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/io/ascii/tests/common.py#L83-L84
- [x] https://github.com/astropy/astropy/pull/15461  https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/modeling/fitting.py#L2060-L2064
- [x] https://github.com/astropy/astropy/pull/15461 https://github.com/astropy/astropy/blob/1ae27848e23910d6f1383ee47e02a53d8d73b3b3/astropy/modeling/fitting.py#L2107-L2111

### Additional context

The procedure of removing deprecated functionality should be as follows:

1. Delete the relevant code, including the `@deprecated` decorator.
2. If the deprecation was mentioned in a docstring then update it accordingly.
3. Remove the associated tests that check that deprecation warnings are raised. A simple way to find them is to [run `pytest`](https://docs.astropy.org/en/latest/development/testguide.html#testing-guidelines), which will reveal the tests that started failing because of the removal of deprecated code. 
4. Remove any imports that are no longer needed. Note that if you follow [`astropy` developer guidelines that explain how to use the `pre-commit` framework](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit) then the unnecessary imports can be detected and removed automatically before or at the time of making the commit.
5. [Add a change log entry](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#add-a-changelog-entry) to the API section of the appropriate sub-package.

### PR DESCRIPTION
**Pull request by Amenaske**
### Description

This pull request is to address the removal of deprecated code in the modeling subpackage for the fitting module.

Address one of #15455

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
Thanks! Still need:

* Change log
* Update usage in `docs/modeling/new-fitter.rst`

**Comment by pllim** [comment]
I also think this needs to be done together with `_fitter_to_model_params`.

File: astropy/modeling/fitting.py [hunk 1]
```diff
@@ -2104,14 +2104,6 @@ def model_to_fit_params(model):
...
 
 
-@deprecated(
-    since="5.1",
-    message="private method: _model_to_fit_params has been made public now",
-)
-def _model_to_fit_params(model):
-    return model_to_fit_params(model)
-
-
```
Comment (WilliamJamieson) [inline astropy/modeling/fitting.py:2114]: As @pllim pointed out, the `_filter_to_model_params` function should be removed at the same time as this. They are a matched pair of related functions.

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]

**Comment by pllim** [comment]
@Amenaske , are you interested to address the review comments? Given that our v6.0 feature freeze is fast approaching, if you do not have time or interest to wrap this up in a few days, I can take over. Please let me know. Thank you!

**Comment by Amenaske** [comment]
Yes, I'll address the review comment.

**Review by WilliamJamieson — APPROVED** [review]

**Review by WilliamJamieson — DISMISSED** [review]
The docs still need to be updated.

**Comment by pllim** [comment]
Since we're fast approaching feature freeze, I will just clean this one up and wrap it up. Hope you don't mind. Thanks!

**Review by pllim — APPROVED** [review]
Thanks!

**Pull request merged**